frontera/tests/mocks/components.py

```diff
@@ -0,0 +1,149 @@
+from frontera.core.components import Backend, Middleware, CanonicalSolver
+
+
+class FakeBackend(Backend):
+
+    def __init__(self, manager):
+        self.seeds = []
+        self.links = []
+        self.responses = []
+        self.requests = []
+        self.errors = []
+        self.lists = [self.seeds, self.responses, self.links, self.errors]
+        self._started = False
+        self._stopped = False
+        self._finished = False
+
+    @classmethod
+    def from_manager(cls, manager):
+        return cls(manager)
+
+    def frontier_start(self):
+        self._started = True
+
+    def frontier_stop(self):
+        self._stopped = True
+
+    def finished(self):
+        return self._finished
+
+    def add_seeds(self, seeds):
+        for seed in seeds:
+            self.seeds.append(seed)
+
+    def page_crawled(self, response, links):
+        for link in links:
+            self.links.append(link)
+        self.responses.append(response)
+
+    def put_requests(self, requests):
+        for request in requests:
+            self.requests.append(request)
+
+    def get_next_requests(self, max_next_requests, **kwargs):
+        lst = []
+        for i in range(max_next_requests):
+            if self.requests:
+                lst.append(self.requests.pop())
+        return lst
+
+    def request_error(self, request, error):
+        self.errors.append((request, error))
+
+
+class FakeMiddleware(Middleware):
```

Comment:
Is it possible to unify FakeMiddleware and FakeBackend? They look quite similar.